### ISSUE TITLE
[Compatibility Version] Configuration APIs Lifecycle

### ISSUE THREAD

**Issue by Jefftree** [kubernetes/kubernetes#127972]
With compatibility version, we check emulation version in deciding whether to serve a resource: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/deleted_kinds.go#L119. Thus, if a resource was introduced in a later version than the emulation version, the resource will be disabled. 

Kubernetes also has other API types that are not directly served but used for configuration files (under the apiserver group). One example is AuthorizationConfiguration (currently in beta). Adding a new version involves directly adding to the `install.go` script, bypassing any compatibility version checks and allowing it to be used in the scheme for conversion/decoding to the internal type: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/apiserver/load/load.go#L73. 

This type is planned to go GA in 1.32, which means that configuration files are able to use the v1 type when declaring their `AuthorizationConfiguration`, even when emulating a previous version.

Is it fine to allow these new configuration types to be used during emulation? I see these similar to adding a new CLI flag, we generally don't version apiserver CLI flag options. Furthermore these types are not persisted to etcd so we have no prerelease lifecycle information on them. 

/cc @jpbetz @aaron-prindle

**Comment by jpbetz** [comment]
> Is it fine to allow these new configuration types to be used during emulation? I see these similar to adding a new CLI flag, we generally don't version apiserver CLI flag options. Furthermore these types are not persisted to etcd so we have no prerelease lifecycle information on them.

I agree. This cluster admin facing in the same way CLI flags are. We did not originally conceive of compatibility version as being enforce in this direction, and I don't consider it in scope for the enhancement.

Curious if anyone disagrees.

**Comment by sftim** [comment]
There are also APIs that aren't served via HTTP, but are used elsewhere, eg [image policy](https://kubernetes.io/docs/reference/config-api/imagepolicy.v1alpha1/)
